### PR TITLE
Feature/173 remove menus part 2

### LINKED ISSUE #173: Remove menus: Part 2
Add endpoints to migrate:
- [x] menu records into documents
- [x] update 'image-tile' type documents pointing to menus
- [x] return `deleteMenus` logs as a response of the corresponding endpoint
- [x] use technical user (#187) to run the migration

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes #173 

To test with Postman:

- make a `POST http://localhost:3000/api/v1/users/login` request with body JSON set to `{ "username": "<username>", "password": "<password>"` for a user with `super-user` role
- you will be authenticated in Postman, session cookie `SID` will be set
- make a `DELETE http://localhost:3000/api/v1/menus` request

### CODE REVIEW

File: src/server/menu-controller.js [hunk 1]
```diff
@@ -15,11 +16,12 @@ function visitMenuNodes(nodes, cb) {
...
 
 class MenuController {
-  static get inject() { return [MenuService, DocumentService, ClientDataMapper, PageRenderer]; }
+  static get inject() { return [MenuService, DocumentService, UserService, ClientDataMapper, PageRenderer]; }
 
-  constructor(menuService, documentService, clientDataMapper, pageRenderer) {
+  constructor(menuService, documentService, userService, clientDataMapper, pageRenderer) {
     this.menuService = menuService;
     this.documentService = documentService;
+    this.userService = userService;
```
Comment (laurentiu-ilici) [inline src/server/menu-controller.js:24]: This doesn't seem to be used anywhere in this file, is it really needed?

File: src/services/menu-service.js [hunk 2]
```diff
@@ -79,6 +88,156 @@ class MenuService {
...
   }
+
+  async createMarkdownContentFromNodes(nodes, level, lines = []) {
+    for (const node of nodes) {
+      if (node.title) {
+        lines.push(`###${'#'.repeat(level)} ${node.title}`);
+        lines.push('');
+      }
+
+      if (node.documentKeys.length) {
```
Comment (laurentiu-ilici) [inline src/services/menu-service.js:99]: Could documentKeys be undefined? safer would be node.documentKeys?.length

File: src/services/menu-service.js [hunk 3]
```diff
@@ -79,6 +88,156 @@ class MenuService {
...
+        for (const documentKey of node.documentKeys) {
+          const document = await this.documentService.getDocumentByKey(documentKey);
+          const documentUrl = `/${document.namespace}/${encodeURIComponent(document.slug)}`;
+          lines.push(`* [${document.title}](${documentUrl})`);
+        }
+        lines.push('');
+      }
+
+      if (node.children.length) {
```
Comment (laurentiu-ilici) [inline src/services/menu-service.js:108]: Okay I guess these can't be undefined

File: src/services/menu-service.js [hunk 4]
```diff
@@ -79,6 +88,156 @@ class MenuService {
...
+
+    try {
+      lock = await this.documentLockStore.takeLock(documentKey);
+      const documentRevisions = await this.documentService.getAllDocumentRevisionsByKey(documentKey);
+
+      documentRevisions.forEach(documentRevision => {
+        documentRevision.sections
+          .filter(section => section.content?.tiles)
+          .forEach(section => section.content?.tiles
+            .filter(tile => tile.link?.type === 'menu')
```
Comment (laurentiu-ilici) [inline src/services/menu-service.js:185]: Well, this was hard to read, but I guess we are deleting this soon :))

File: src/services/menu-service.js (hunk 2 above)
Comment (adrianaluchian) [reply src/services/menu-service.js:99]: They are always empty string if there are no keys, tested on the production dump

File: src/server/menu-controller.js [hunk 5]
```diff
@@ -86,6 +88,10 @@ class MenuController {
       const menu = await this.menuService.saveMenu({ menu: req.body, user: req.user });
       return res.send({ menu });
     });
+    router.delete('/api/v1/menus', [needsPermission(permissions.MIGRATE_DATA), jsonParser], async (req, res) => {
```
Comment (ahelmberger) [inline src/server/menu-controller.js:91]: No need for JSON parser here (although it doesn't really screw it up) :-)

Comment (adrianaluchian) [reply src/server/menu-controller.js:91]: What parser? 🙄 😝

**Review by laurentiu-ilici — APPROVED** [review]

**Pull request merged**